### PR TITLE
update contributor guide, add release scripts

### PR DESCRIPTION
**Pull request by trusktr**

### CODE REVIEW

File: docs/contributor_guide.md [hunk 1]
```diff
@@ -8,10 +8,14 @@ If you are looking for documentation on _how to use_ the library, the [user guid
...
-- [Continuous integration](#continuous-integration)
-- [Release process](#release-process)
+- [tween.js contributor guide](#tweenjs-contributor-guide)
+  - [Developer requirements](#developer-requirements)
+  - [Testing](#testing)
+    - [Unit tests](#unit-tests)
+    - [Code style and lint tests](#code-style-and-lint-tests)
+    - [Other types of tests](#other-types-of-tests)
+  - [Continuous integration](#continuous-integration)
+  - [Release process](#release-process)
```
Comment (trusktr) [inline docs/contributor_guide.md:18]: VS Code automatically updated this (I didn't have to do it, yay)

File: scripts/write-version.js [hunk 2]
```diff
@@ -1,9 +1,16 @@
...
+function handleError(error) {
+	if (error) {
+		console.error(error)
+		process.exit(1)
+	}
+}
+
 fs.open('./src/Version.ts', 'w', (error, fd) => {
-	if (error) process.exit(1)
+	handleError(error)
```
Comment (trusktr) [inline scripts/write-version.js:12]: Better to log some output to the terminal if something goes wrong.

File: docs/contributor_guide.md [hunk 3]
```diff
@@ -118,137 +106,14 @@ There's an [open issue](https://github.com/tweenjs/discuss/issues/3) to track wo
...
-Now we need to enable the project in Travis CI, so make sure you have an account there and are logged in.
-
-Enable the project in Travis (if you're the maintainer) or ask the maintainer to enable it, in the [Travis settings page](https://travis-ci.org/profile/).
-
-Click on the cog near to the project name to configure some options.
-
-Scroll down until you see _Environment Variables_.
-
-Add tokens for `GH_TOKEN` and `NPM_TOKEN`. Make sure both variables are hidden: `Display value in build log` should be `Off`.
+Currently the release process is manual.
```
Comment (trusktr) [inline docs/contributor_guide.md:115]: Removed the semantic-release/Travis description, as we're not using it currently. Maybe we can add semantic-release back on top of GitHub actions (later PR).

File: package.json [hunk 4]
```diff
@@ -36,7 +36,11 @@
 		"test-lint": "npm run prettier -- --check && eslint 'src/**/*.ts'",
 		"lint": "npm run prettier -- --write && eslint 'src/**/*.ts' --fix",
 		"prettier": "prettier './**/*.{js,ts,md,json,html,css}'",
-		"prepare": "npm run build"
+		"prepare": "npm run build",
+		"version": "npm test && git add .",
+		"release:patch": "npm version patch --message 'v%s' && npm publish && git push --follow-tags",
+		"release:minor": "npm version minor --message 'v%s' && npm publish && git push --follow-tags",
+		"release:major": "npm version major --message 'v%s' && npm publish && git push --follow-tags"
```
Comment (trusktr) [inline package.json:43]: We run one of the `release:*` scripts to publish to NPM. This builds and tests before publishing and finally pushing the updated files (including `dist/` and `src/Version.ts` files which are currently tracked in the repo) to GitHub along with the version tag.

Maybe we should remove publishing of `dist/` to GitHub, but that was out of scope of this PR.

Comment (trusktr) [reply package.json:43]: Note, I tested these scripts using `npm publish --dry-run` so that npm wouldn't actually publish anything.

**Comment by dalisoft** [comment]
Maybe we add commit lint? Add files lint bu lint-staged and husky?

**Review by dalisoft — COMMENTED** [review]
Would be nice to see more linters and strict checking + TypeScript checking. 
Like in order
- typescript
- eslint
- prettier
- commit-lint
- lint-staged

And then commit or publish to npm?

**Comment by trusktr** [comment]
> Maybe we add commit lint? Add files lint bu lint-staged and husky?

Yeah, that'd be nice, and also restoring semantic-release. We can make new PRs for those.

**Comment by trusktr** [comment]
For now, I did this because there was no obvious way to publish. Got the TypeScript export types fix coming next.

**Pull request merged**